OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -440,10 +448,12 @@ protected override MovePart OnComplete(Actor self, Mobile mobile, Move parent)
 					parent.path.Add(nextCell.Value.First);
 				}
 
+				var toPos = mobile.ToCell.Layer == 0 ? map.CenterOfCell(mobile.ToCell) :
+					((ActorMap)self.World.ActorMap).CustomMovementLayers[mobile.ToCell.Layer].CenterOfCell(mobile.ToCell);
```

Comment:
You make this cast a lot to get at `CustomMovementLayers` - is there a reason to not just add that to the interface?